docs/aws/audit/kmsmonitoring/rules/secret_rotation_interval.mdx

```diff
@@ -23,6 +23,237 @@ HITRUST, AWSWAF, PCIDSS, ISO27001
+       --rotation-rules AutomaticallyAfterDays=30
+   ```
+
+4. **Verify Rotation Configuration:**
+   To ensure that the rotation configuration is correctly set, you can describe the secret and check its rotation rules using the `describe-secret` command.
+
+   ```sh
+   aws secretsmanager describe-secret \
+       --secret-id MySecretName
+   ```
+
+   Look for the `RotationEnabled` and `RotationRules` fields in the output to verify that rotation is enabled and configured correctly.
+
+By following these steps, you can ensure that your secrets in AWS Secrets Manager are rotated frequently, thereby enhancing the security of your secrets.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of not rotating Secrets Manager secrets frequently in AWS KMS using Python scripts, you can follow these steps:
+
+### 1. **Set Up AWS SDK for Python (Boto3)**
+First, ensure you have the AWS SDK for Python (Boto3) installed. You can install it using pip if you haven't already:
+
+```bash
+pip install boto3
+```
+
+### 2. **Create a Python Script to Enable Automatic Rotation**
+You can create a Python script to enable automatic rotation for your secrets in AWS Secrets Manager. Below is an example script:
+
+```python
+import boto3
+from botocore.exceptions import ClientError
+
+# Initialize a session using Amazon Secrets Manager
+client = boto3.client('secretsmanager')
+
+def enable_secret_rotation(secret_id, rotation_lambda_arn):
+    try:
+        response = client.rotate_secret(
+            SecretId=secret_id,
+            RotationLambdaARN=rotation_lambda_arn,
+            RotationRules={
+                'AutomaticallyAfterDays': 30  # Rotate every 30 days
+            }
+        )
+        print(f"Rotation enabled for secret: {secret_id}")
+    except ClientError as e:
+        print(f"Error enabling rotation for secret {secret_id}: {e}")
+
+# Replace 'your-secret-id' and 'your-lambda-arn' with your actual secret ID and Lambda ARN
+secret_id = 'your-secret-id'
+rotation_lambda_arn = 'your-lambda-arn'
+
+enable_secret_rotation(secret_id, rotation_lambda_arn)
+```
+
+### 3. **Schedule the Script to Run Periodically**
+To ensure that the script runs periodically, you can use a scheduling tool like cron (on Unix-based systems) or Task Scheduler (on Windows). This will help you to regularly check and enforce the rotation policy.
+
+### 4. **Monitor and Alert on Rotation Status**
+You can set up CloudWatch Alarms to monitor the rotation status of your secrets. If a secret is not rotated within the expected time frame, an alert can be triggered. Below is an example of how you can create a CloudWatch alarm using Boto3:
+
+```python
+import boto3
+
+# Initialize a session using Amazon CloudWatch
+cloudwatch = boto3.client('cloudwatch')
+
+def create_rotation_alarm(secret_id):
+    try:
+        response = cloudwatch.put_metric_alarm(
+            AlarmName=f'SecretRotationAlarm-{secret_id}',
+            MetricName='RotationAge',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>